### PR TITLE
Feature/search 1654

### PR DESCRIPTION
**Pull request by eliaporciani**

### CODE REVIEW

File: src/main/java/org/alfresco/repo/domain/node/Node.java [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
+ * 
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
```
Comment (aborroy) [inline src/main/java/org/alfresco/repo/domain/node/Node.java]: Previous license was wrong? It looks the same to me...

File: src/main/java/org/alfresco/repo/domain/node/NodeEntity.java [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
 package org.alfresco.repo.domain.node;
```
Comment (aborroy) [inline src/main/java/org/alfresco/repo/domain/node/NodeEntity.java]: Previous license was wrong? It looks the same to me...

File: src/main/java/org/alfresco/repo/domain/node/NodeEntity.java [hunk 3]
```diff
@@ -54,8 +54,9 @@
     private Long localeId;
     private Long aclId;
     private TransactionEntity transaction;
-    private AuditablePropertiesEntity auditableProperties;
+    private AuditablePropertiesEntity auditableProperties;
```
Comment (aborroy) [inline src/main/java/org/alfresco/repo/domain/node/NodeEntity.java]: Replacing a tab by spaces?

File: src/main/java/org/alfresco/repo/domain/node/NodeEntity.java [hunk 4]
```diff
@@ -90,7 +91,7 @@ public NodeEntity()
         this.localeId = node.getLocaleId();
         this.aclId = node.getAclId();
         this.transaction = node.getTransaction();
-        this.auditableProperties = node.getAuditableProperties();
+        this.auditableProperties = node.getAuditableProperties();
```
Comment (aborroy) [inline src/main/java/org/alfresco/repo/domain/node/NodeEntity.java]: Replacing a tab by spaces?

File: src/main/java/org/alfresco/repo/domain/node/NodeEntity.java [hunk 5]
```diff
@@ -139,7 +140,7 @@ public String toString()
           .append(", localeId=").append(localeId)
           .append(", aclId=").append(aclId)
           .append(", transaction=").append(transaction)
-          .append(", auditProps=").append(auditableProperties)
+          .append(", auditProps=").append(auditableProperties)
```
Comment (aborroy) [inline src/main/java/org/alfresco/repo/domain/node/NodeEntity.java]: More tabs-spaces replacing?

File: src/main/java/org/alfresco/repo/index/shard/ShardRegistry.java [hunk 6]
```diff
@@ -1,52 +1,68 @@
...
+ * 
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
```
Comment (aborroy) [inline src/main/java/org/alfresco/repo/index/shard/ShardRegistry.java]: Previous license was wrong? It looks the same to me...

File: src/main/java/org/alfresco/repo/solr/NodeParameters.java [hunk 7]
```diff
@@ -1,28 +1,28 @@
...
+ * 
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
```
Comment (aborroy) [inline src/main/java/org/alfresco/repo/solr/NodeParameters.java]: Previous license was wrong? It looks the same to me...

File: src/main/java/org/alfresco/repo/solr/NodeParameters.java [hunk 8]
```diff
@@ -54,10 +54,11 @@
...
     
     private Set<QName> includeAspects;
-    private Set<QName> excludeAspects;
-    
-    private String shardProperty;
+    private Set<QName> excludeAspects;
     
+    private String shardProperty;
+    private String coreName;
+
```
Comment (aborroy) [inline src/main/java/org/alfresco/repo/solr/NodeParameters.java]: This change only includes "coreName" field, but it changes other lines... Again the spaces-tab fighting?

File: src/main/resources/alfresco/ibatis/org.alfresco.repo.domain.dialect.Dialect/solr-common-SqlMap.xml [hunk 9]
```diff
@@ -115,12 +116,13 @@
...
-            node.acl_id         as acl_id,
-            store.protocol      as protocol,
-            store.identifier    as identifier,
+            node.type_qname_id      as type_qname_id,
+            node.transaction_id     as txn_id,
+            txn.commit_time_ms      as txn_commit_time_ms,
+            node.uuid               as uuid,
+            node.acl_id             as acl_id,
+            store.protocol          as protocol,
+            store.identifier        as identifier,
```
Comment (aborroy) [inline src/main/resources/alfresco/ibatis/org.alfresco.repo.domain.dialect.Dialect/solr-common-SqlMap.xml]: This change includes only "txn.commit_time_ms" in the query, but it looks more complex due to re-indentation... Probably all these changes can be adjusted to reflect only the real change.

File: src/test/java/org/alfresco/repo/search/impl/solr/TestNode.java [hunk 10]
```diff
@@ -1,28 +1,28 @@
...
+ * 
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
```
Comment (aborroy) [inline src/test/java/org/alfresco/repo/search/impl/solr/TestNode.java]: Previous license was wrong? It looks the same to me...

File: src/test/java/org/alfresco/repo/search/impl/solr/TestNode.java [hunk 11]
```diff
@@ -131,12 +131,18 @@ public TransactionEntity getTransaction()
...
+    @Override
+    public String getShardKey()
+    {
+        return null;
+    }
+
+    @Override
+    public Integer getExplicitShardId()
+    {
+        return null;
```
Comment (aborroy) [inline src/test/java/org/alfresco/repo/search/impl/solr/TestNode.java]: Only including "getExplicitShardId" method, but changes are affecting other parts of the code.

File: src/main/java/org/alfresco/repo/domain/node/NodeEntity.java [hunk 12]
```diff
@@ -334,18 +335,30 @@ public synchronized void setAuditableProperties(AuditablePropertiesEntity audita
...
+        this.shardKey = shardKey;
+    }
+
+    public synchronized void setExplicitShardId(Integer explicitShardId)
+    {
+         this.explicitShardId = explicitShardId;
+    }
+
+
+
```
Comment (aborroy) [inline src/main/java/org/alfresco/repo/domain/node/NodeEntity.java]: Only including "getExplicitShardId" and "setExplicitShardId" methods, but changes are affecting other parts of the code.

File: src/main/java/org/alfresco/repo/solr/NodeParameters.java [hunk 13]
```diff
@@ -181,15 +182,25 @@ public void setIncludeAspects(Set<QName> includeAspects)
...
+    }
+
+    public String getCoreName() {
+        return this.coreName;
+    }
+
+    public void setCoreName(String coreName){
+        this.coreName = coreName;
+    }
+
```
Comment (aborroy) [inline src/main/java/org/alfresco/repo/solr/NodeParameters.java]: Only including "getCoreName" and "setCoreName" methods, but changes are affecting other parts of the code.

**Comment by eliaporciani** [comment]
There is a new line character problem. The files contained mixed characters for that and they have been modified to use the same character uniformly.

**Pull request merged**